kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigMapWatcher.java

```diff
@@ -192,8 +197,16 @@ private void refreshEnvironment() {
         }
     }
 
+    /**
+     * Process {@link EventType#ERROR} events, unconditionally restarting the watch.
+     *
+     * @see <a href="https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes">Efficient detection of changes</a>
+     */
     private void processConfigMapErrored(ConfigMapWatchEvent event) {
         LOG.error("Kubernetes API returned an error for a ConfigMap watch event: {}", event.toString());
+        KubernetesConfigurationClient.getPropertySourceCache().clear();
+        refreshEnvironment();
```

Comment:
Wouldn't this mean that configuration users would observe an empty configuration every time this happens?

That seems at the very least unnecessary and possibly dangerous if some components don't expect their configuration to disappear without warning.

Reply:
Fortunately nope: the first done in `refreshEnvironment` is to gather the new environment, then compare it with the old one in order to generate a diff. Only then, the event is broadcasted:
```
        changes = environment.refreshAndDiff(); // captures the old env, updates it to the new one, and computes changes out of them
[...]
        eventPublisher.publishEvent(new RefreshEvent(changes)); // then signals changes
```

Reply:
I forgot that we use the empty cache as a trigger for re-listing the ConfigMaps.